sdk/scheduler/src/main/java/com/mesosphere/sdk/framework/FrameworkConfig.java

```diff
@@ -281,4 +280,26 @@ public String getWebUrl() {
   public String toString() {
     return ReflectionToStringBuilder.toString(this);
   }
+
+  /**
+   * Returns the non-namespaced role used in earlier versions.
+   * Can be compared with {@link #getRole()} to determine which role
+   * the current footprint has been deployed with.
+   */
+  public String getNonNamespacedRole() {
+    return SchedulerUtils.withEscapedSlashes(frameworkName) + DEFAULT_ROLE_SUFFIX;
+  }
+
+  /**
+   * Returns the namespaced role derived from the framework name.
+   */
+  public Optional<String> getNamespacedRole() {
+    String[] marathonGroups = getNonNamespacedRole().split("__");
```

Comment:
it would be nice to get `__` somehow from `SchedulerUtils` 
One way is to make `SLASH_REPLACEMENT` in that class public and use it here, or create a new `public static` method. anyway this is a nit.